kernel/kernel-api/src/main/java/io/delta/kernel/types/StructField.java

```diff
@@ -175,6 +175,16 @@ public StructField withTypeChanges(List<TypeChange> typeChanges) {
     return new StructField(name, dataType, nullable, metadata, typeChanges);
   }
 
+  /**
+   * Creates a copy of this StructField with the specified data type.
+   *
+   * @param newType The new type to use in the StructField.
+   * @return A new StructField with the same properties but with the specified data type.
+   */
+  public StructField withDataType(DataType newType) {
+    return new StructField(name, newType, nullable, metadata, typeChanges);
```

Comment:
For my understanding: should typeChanges needs to update as well?

Reply:
Can you add a note to indicate that "type changes externally and update as a whole"?